src/harness/harness.ts

```diff
@@ -809,8 +809,34 @@ namespace Harness {
         const lineFeed = "\n";
 
         export const defaultLibFileName = "lib.d.ts";
-        export const defaultLibSourceFile = createSourceFileAndAssertInvariants(defaultLibFileName, IO.readFile(libFolder + "lib.core.d.ts"), /*languageVersion*/ ts.ScriptTarget.Latest);
-        export const defaultES6LibSourceFile = createSourceFileAndAssertInvariants(defaultLibFileName, IO.readFile(libFolder + "lib.core.es6.d.ts"), /*languageVersion*/ ts.ScriptTarget.Latest);
+        export const defaultLibSourceFile = createSourceFileAndAssertInvariants(defaultLibFileName, IO.readFile(libFolder + "lib.es5.d.ts"), /*languageVersion*/ ts.ScriptTarget.Latest);
+        export const defaultES6LibSourceFile = createSourceFileAndAssertInvariants(defaultLibFileName, IO.readFile(libFolder + "lib.es6.d.ts"), /*languageVersion*/ ts.ScriptTarget.Latest);
+
+        export const libFileNameSourceFileMap: ts.Map<ts.SourceFile> = {
+            "lib.es5.d.ts": createSourceFileAndAssertInvariants("lib.es5.d.ts", IO.readFile(libFolder + "lib.es5.d.ts"), ts.ScriptTarget.Latest),

```

Comment:
i do not think we need to read them all on the first time. we only need to read and cache the ones we are asked to get. so you will need to change defaultLibSourceFile et al into a getLibSourceFile(...) and that will do the trick. no need to list them all either.